docs/src/content/docs/book/functions.mdx

```diff
@@ -114,6 +116,175 @@ native storeUint(s: Builder, value: Int, bits: Int): Builder;
+    definePlusPlus();
+    definePlusPlus();
+    definePlusPlus();
+
+    // Now, let's use the ++
+    return inc(41); // 42
+}
+```
+
+:::caution
+
+  Prefer to use _only_ [TVM instructions][tvm-instructions] only and _never_ define new Fift words or use existing ones, since it's very easy to confuse compile-time and run-time execution and forget which stack is used where and when — Fift's or TVM's.
+
+:::
+
+It is not necessary to enclose TVM instructions in double quotes. On the contrary, they are interpreted by Fift as strings, which is probably _not_ what you want:
+
+```tact
+// Puts the string "MYCODE" on Fift's stack at compile-time,
+// where it gets discarded even before the compute phase starts
+asm fun wrongMyCode() { "MYCODE" }
+
+// Invokes the TVM instruction MYCODE during the compute phase,
+// which returns the contract code as a Cell
+asm fun myCode(): Cell { MYCODE }
+```
+
+The syntax for parameters and return values is the same as for other function kinds, but there is one caveat — argument values are pushed to the stack before the function body is executed, and return values are what's left on the stack afterward.
+
+Since the bodies of `asm{:tact}` functions do not contain Tact statements, any direct references to parameters in function bodies will be recognized as Fift words, which can easily lead to very obscure error messages.
+
+```tact
+/// Simply returns back the value of `x`
+asm fun identity(x: Int): Int { }
+
+/// COMPILATION ERROR!
+/// You could've thought that you're providing the value of `boc` to the function
+/// and receiving it back, but actually there's a builtin Fift word `boc` which is
+/// being used instead.
+/// The `boc` would not be recognized as a parameter
+/// even if such word didn't exist in Fift!
+asm fun bocchiThe(boc: Cell): Cell { boc }
+
+/// Loads a signed `l`-bit integer from Slice `s`,
+/// and returns it with the remainder of `s`
+asm fun sliceLoadInt(s: Slice, l: Int): IntSlice { LDIX }
+//                   ↑         ↑
+//                   |         Placed on the stack last
+//                   Placed on the stack first
+
+// Used to map onto values placed by LDIX on the stack
+struct IntSlice { a: Int; b: Slice }
+```
+
+### Arrangements {#asm-arrangements}
+
+Sometimes it's useful to change the order of arguments pushed to the stack or the order of return values. You can do that with `asm{:tact}` arrangements in the following manner:
+
+```tact
+// Changing the order of arguments to match the STDICT signature
+asm(c self) extends fun asmStoreDict(self: Builder, c: Cell?): Builder { STDICT }
+
+// Changing the order of return values of LDVARUINT16,
+// capturing only the 2nd one as the return value of the whole function
+asm(-> 1 0) extends mutates fun asmLoadCoins(self: Slice): Int { LDVARUINT16 }
+
+// Changing the order of argument and return values
+asm(self len -> 1 0) extends fun asmLoadInt(self: Slice, l: Int): SliceInt { LDIX }
```

Comment:
Identifier `len` in the arrangement does not occur in the argument list of the function. Also, the arrangement is not really rearranging the function's arguments: they are in the same order. So the comment above the function: "Changing the order of argument and return values" is misleading.